revised Curated Item docs

MoltonMontro opened the pull request:
- Note about Skin Base Textures for items.
- Couple new images.
- Rewrote the "Curated Items" doc.

No mentions of mythical effects or vehicle skins, since these aren't relevant quite at the moment. When peer-reviewing, could you pay particular note to the updated guidelines?

Also – the og Mythical Effect Placement section says that the Effect is placed in the Item.prefab. But I think this is actually the _other_ prefab (e.g., Mask.prefab). If so, is there a good term to describe that prefab (since its name changes per item type)?

_Note to self: once a PR is accepted, I'll update the categories on a few wiki articles that are linked in the revised docs._

MoltonMontro committed:
revised Curated Item docs

- Note about Skin Base Textures for items.
- Couple new images.
- Rewrote the "Curated Items" doc.

SDGNelson commented:
Well worded and good improvements, thanks!

A few adjustments I suggest:

- Since we don't accept mythical effects through the curated workshop yet, I think "such as skins, cosmetics, and mythical effects" should remove "and mythical effects" for now.

- The guidelines are nice and clear. I think we can remove 8.2 now, however, as that ship has sailed - Escalation being the final nail in the coffin. Community sentiment seems to be fine with it.

- Good catch about the Effect transform in items: It should actually be both prefabs! Maybe the other one could be referred to as the something like: "the clothing prefab (i.e., Backpack.prefab, Glasses.prefab, Hat.prefab, Mask.prefab, Vest.prefab)"?

MoltonMontro committed:
further revisions

- fixed Ref anchor
- removed mention of Mythical Effects as a curated item type (no real support/docs yet)
- tweaked guidelines further
- corrected info about Effect transform

MoltonMontro commented:
Latest commit revises based on ^^ feedback.

MoltonMontro committed:
trim whitespace
